TrigScint/include/TrigScint/Firmware/objdef.h

```diff
@@ -62,55 +66,70 @@ inline void cpyHit(Hit& c1, Hit& c2) {
 }
 
 struct Cluster {
-  Hit Seed;
-  Hit Sec;
-  ap_int<11> Cent;
+  Hit Seed{};
+  Hit Sec{};
+  ap_int<12> Cent{};
   // int nhits, mID, SeedID;
   // float CentX, CentY, CentZ, Amp, Time, TrigTime;
 };
+
 inline void clearClus(Cluster& c) {
   clearHit(c.Seed);
   clearHit(c.Sec);
-  c.Cent = (ap_int<11>)(0);  // clearHit(c.For);
+  c.Cent = (ap_int<12>)(0);  // clearHit(c.For);
 }
+
 inline void calcCent(Cluster& c) {
-  if (c.Seed.Amp > 0) {
-    c.Cent = (ap_int<12>)(10. *
-                          ((float)(c.Seed.Amp * c.Seed.bID +
-                                   c.Sec.Amp * c.Sec.bID)) /
-                          ((float)(c.Seed.Amp + c.Sec.Amp)));
-  } else {
+  // Check if Seed and Sec amplitudes are valid
+  if (c.Seed.Amp <= 0 || c.Sec.Amp <= 0) {
+    c.Cent = (ap_int<12>)(0);
+    return;
+  }
+
+  if (c.Seed.bID < 0 || c.Sec.bID < 0) {
     c.Cent = (ap_int<12>)(0);
+    return;
   }

```

Comment:
@rodwyer100 can you confirm that these checks make sense?

Reply:
Those checks are fine; I think there is a cut on the amplitude of clusters in tracking so this would not affect things.